app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }
```

Comment:
Sometimes these active record hooks can bite you, but this feels ok.

Reply:
This is a good point. Where do you recommend I move the .downcase to? I don't think it would go in the validates section... but I'm not really sure where else to put it. Unless I'm overlooking something which I'm notoriously good at doing, unfortunately.

Reply:
Sorry, maybe I wasn't so clear. I do think this is an ok use of these hooks. Another way you could consider would be to do this in a custom save or update method. But that seems kind of overkill and I'd just leave this here.

I mostly wanted to let you know when you shouldn't use these hooks.

Reply:
Oh! Good to know. Thank you for this. :)